Content.Server/Chat/Commands/DeleteChatMessageCommand.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Content.Server.Administration;
+using Content.Server.Chat.V2.Repository;
+using Content.Shared.Administration;
+using Robust.Shared.Console;
+
+namespace Content.Server.Chat.Commands
+{
+    [AdminCommand(AdminFlags.Admin)]
+    internal sealed class DeleteChatMessageCommand : IConsoleCommand
+    {
+        public string Command => "delmsg";
+        public string Description => "Delete a specific chat message";
+        public string Help => "delmsg <id>";
+
+        public void Execute(IConsoleShell shell, string argStr, string[] args)
+        {
+            var repo = IoCManager.Resolve<ChatRepository>();
+
+            if (!uint.TryParse(args[0], out var result))
+            {
+                shell.WriteError("can't delete chat message: invalid number argument");
+
+                return;
+            }
+
+            repo.Delete(result);
+        }
+    }
```

Comment:
1. public
2. needs localisation for description + help.
3. This should probably be a toolshedcommand
4. Put the dependency as a field with the attribute.

Reply:
Not a huge urgency atm as this code is hooked up to nothing but sure

On 4, doing what you and SC say and putting this as a property of the class causes the IoCManager to throw its toys out of the pram during the testing suite, as ChatRepository isn't registered somewhere it needs to be.

Reply:
Resolved. On 4, calling `Resolve` to grab the dep is how every other command in this folder does it